src/main/java/Domain/Role.java

```diff
@@ -0,0 +1,237 @@
+        return _permissions.contains(Permission.OWNER);
+    }
+
+    
+    /**
+     * Check if this role has a specific permission.
+     * @param p permission to check.
+     * @return boolean that represents if the role has this permission.
+     */
+    public boolean hasPermission(Permission p){
+        if(p == null){
+            return false;
+        }
+        return _permissions.contains(p);
+    }
+
+    /**
+     * Check if this role has at least one permission of the given set.
+     * @param permissions permission set.
+     * @return boolean that represents if the role has at least one permission from the set.
+     */
+    public boolean hasAtLeastOnePermission(Set<Permission> permissions){
+        if(permissions == null || permissions.isEmpty()){
+            return false;
+        }
+        permissions.retainAll(_permissions);
+        return !permissions.isEmpty();
+    }
+
+    /**
+     * Check if this role has at all permissions of the given set.
+     * @param permissions permission set.
+     * @return boolean that represents if the role has all permissions from the set.
+     */
+    public boolean hasAllPermissions(Set<Permission> permissions){
+        if(permissions == null || permissions.isEmpty()){
+            return false;
+        }
+        return !permissions.retainAll(_permissions);
+    }
+
+    //TODO: What happens if founder decides to change owner to regular manager?
+    /**
+     * Add permissions to this role.
+     *
+     * @param permission the permissions to add.
+     * @implNote If the role already has this permission its ok.
+     * @throws RoleException 
+     */
+    public void addPermissions(String username, Set<Permission> permissions) throws RoleException {
+        if(isFounder() || isOwner()){
+            throw new RoleException("Username is a founder of owner. No need to manage permissions.");
+        }
+        if(!username.equals(_appointedBy)){
+            throw new RoleException("Only the role that appointed this user can change permission.");
+        }
+        if(permissions.contains(Permission.OWNER) || permissions.contains(Permission.FOUNDER)){
+            throw new RoleException("Cannot add owner or founder permissions.");
+        }
+        _permissions.addAll(permissions);
+    }
+
+
+    /**
+     * Remove a permission from this role.
+     *
+     * @param permission the permissions to remove.
+     * @implNote if role doesn't have permission its ok.
+     * @throws RoleException 
+     */
+    public void deletePermissions(String username, Set<Permission> permissions) throws RoleException {
+        if(isFounder() || isOwner()){
+            throw new RoleException("Username is a founder of owner. No need to manage permissions.");
+        }
+        if(!username.equals(_appointedBy)){
+            throw new RoleException("Only the role that appointed this user can change permission.");
+        }
+        _permissions.removeAll(permissions);
```

Comment:
you cant delete one? have to delete all? when we would like that? :O

Reply:
you delete a given set, if you want to delete one just give one in the set. i think that there will be a page that we can mark the permissions we want to remove and its better like this. same for permission addition.